### PR TITLE
Upgrade xterm to 3.11.0 and remove vendored addon

### PR DESCRIPTION
**Pull request by betatim**
The goal of this PR is to get us to xterm.js 3.11.0.

I had hoped that this was just a matter of editing `package.json` but alas no :(

My guess is that something needs changing in our webpack config because xterm.js is no typescript based??

The state of this PR results in the following error in the console after running `npm run webpack && python3 -m binderhub -f testing/minikube/binderhub_config.py` (with `nvm version` showing v11.7.0)

screen shot 2019-02-13 at 10 19 51

I don't think this has anything to do with the addon because if you remove all reference to the addon you get:
screen shot 2019-02-13 at 10 24 38

So I think something about how we import the xterm stuff needs to change. Could someone point me in a/the right direction for reading material?

### CODE REVIEW

**Comment by betatim** [comment]
I'm going to need an assist from @captainsafia or @yuvipanda (or someone else who understands what we are doing in terms of JS and TS and webpack).

I've updated the code. I think this is how you should use the xterm extensions mechanism. Gives me this error:

screen shot 2019-02-14 at 07 54 20

To setup things:
* checkout this PR
* `npm install` and `npm run webpack`
* `python3 -m binderhub -f testing/localonly/binderhub_config.py` (no need for minikube and friends)

File: binderhub/static/js/index.js [hunk 1]
```diff
@@ -144,7 +143,7 @@ function build(providerSpec, log, path, pathType) {
   image.onStateChange('*', function(oldState, newState, data) {
     if (data.message !== undefined) {
       log.write(data.message);
-      log.fit();
+      fit(log); //log.fit();
```
Comment (betatim) [inline binderhub/static/js/index.js]: Turning this around so that you don't need `applyAddon` is what jupyter lab does as well and seems to work.

**Comment by betatim** [comment]
With the changed way of using `fit()` things are working ... except the fit addon struggles with us hiding the parent HTML element (NaNs start appearing in its calculation of the height and width)

File: binderhub/app.py [hunk 2]
```diff
@@ -444,6 +444,9 @@ def initialize(self, *args, **kwargs):
                 kubernetes.config.load_kube_config()
             self.tornado_settings["kubernetes_client"] = self.kube_client = kubernetes.client.CoreV1Api()
 
+        else:
```
[the remark this answers is not in the document]
Comment (betatim) [reply binderhub/app.py]: I needed this (as a hack) to be able to run in "frontend dev only mode". I didn't spend a lot of time trying to work out why this change or how else one could fix it. Once this PR works we should think about the proper way of fixing this (or punting to a new PR).

**Comment by betatim** [comment]
Just read some [feedback from a workshop that used Binder](https://github.com/alan-turing-institute/the-turing-way/issues/258#issuecomment-469648796) and one comment was "I wouldn't have ever thought of unfolding that logging pane" which makes me wonder: should we have it open by default? 

Less tidy but easier to discover and maybe gives more of an impression of "wow, stuff is happening!".

If we started toggled open I think that would solve the problem here of the plugin being confused about starting from size zero. So we could still have the toggle to hide the logs for those who aren't calmed by seeing them.

Reflecting on my personal usage pattern I almost always unfold the logs to see what is up and I think in the few cases where I don't care about keeping track of progress I'd probably also not care about a busy log scrolling by.

**Comment by yuvipanda** [comment]
I too prefer the logs to be open by default...

**Comment by captainsafia** [comment]
In that case, maybe it is worth exploring having the logs open by default as a way to workaround the issue @betatim mentioned and to provide this to users. Might kill two birds with one stone.

**Comment by manics** [comment]
See https://github.com/jupyterhub/binderhub/pull/1365

**Pull request closed without merging**